pkg/operator/operator.go

```diff
@@ -176,6 +176,10 @@ func NewOperator(ctx context.Context, operator *operator.Operator) (context.Cont
 		launchTemplateProvider,
 	)
 
+	if cidrErr := launchTemplateProvider.ResolveClusterCIDR(ctx); cidrErr != nil {
```

Comment:
This will cause crashing now for anyone who isn't using DescribeCluster, which is a major breaking change. 

The reason this existed before only as an error and was gated for only AL2023 is because we took the stance that users using AL2023 should just give us DescribeCluster permission. However, not everyone is using EKS and certainly not everyone is giving us DescribeCluster. We should shift this back to be gated only by AL2023.

Reply:
Sounds good, add this back into the status controller and gated to be only resolved on AL2023. I also added a comment just to outline why we do it as reconcile loop and not on start-up.

Reply:
Thanks. Adding a comment is a good call, since I'm realizing after having this convo with you that only @jmdeal and I were aware of exactly why this line existed where it did